lib/tests/streamlit/components_test.py

```diff
@@ -241,17 +246,70 @@ def test_json_and_df_args(self):
         self.assertIsNotNone(proto.args_dataframe)
 
     def test_key(self):
-        """Test the 'key' param"""
-        # TODO!
-        pass
+        """Test the 'key' param."""
+        self.test_component(foo="bar", key="baz")
+        proto = self.get_delta_from_queue().new_element.component_instance
+        self.assertEqual("baz", proto.id.split("-")[0])
```

Comment:
This feels like it's testing the wrong thing. We don't care how the protobuf's ID is constructed - what we care about is that its ID remains constant if its key remains constant:

```
# Two components with the same `key` should have the same ID
self.test_component(foo="bar", key="baz")
proto1 = self.get_delta_from_queue().new_element.component_instance

self.test_component(key="baz")
proto2 = self.get_delta_from_queue().new_element.component_instance
self.assertEqual(proto1.id, proto2.id)
```

(You may have to patch some widget-registration guts to get these component creation calls to work without raising a "You already created a `key="baz"` component!" error)

Reply:
Done!